File: exercises/python_playwright/Click/click_playwright.py
from playwright.sync_api import Page


class Click:
    def __init__(self, page: Page):
        """Store the Playwright page and button locator."""
        self.page = page
        self.button = page.locator("#badButton")


    def open(self) -> None:
        """Open the Click exercise page."""
        self.page.goto("http://uitestingplayground.com/click")


    def click_button(self) -> None:
        """Click the target button."""
        self.button.click()

File: exercises/javascript_playwright/click/clickPlaywright.js
export class ClickPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.button = page.locator("#badButton");
    }

    async openPage() {
        await this.page.goto("http://uitestingplayground.com/click");
    }

    async clickButton() {
        await this.button.click();
    }
}
